tensorflow_datasets/core/example_serializer_test.py

```diff
@@ -218,6 +218,39 @@ def test_item_to_tf_feature_string_check(self):
     ):
       example_serializer._item_to_tf_feature(example_item, tensor_info)
 
+  def test_dict_to_tf_example_error_reraise(self):
+    # Test error reraise in _dict_to_tf_example
+    example_data = {'input': [1, 2, 3]}
+    tensor_info = {
+        'input': feature_lib.TensorInfo(
+            shape=(2,),
+            dtype=tf.int64,
+        ),
+    }
+    with self.assertRaisesRegex(
+        ValueError,
+        (
+            '^Error while serializing feature `input`:'
+            ' `TensorInfo\(shape=\(2,\), dtype=tf.int64\)`:(.*)'
```

Comment:
I like that this checks the correct error is raised, but I worry that it's a bit fragile because it depends on the output format of `TensorInfo`. Would just "Error while serializing feature" be sufficient to confirm the error?

Reply:
I think it would be. An alternative, if we want to check that the correct `tensor_info` is included in the error message, is to get the string representation of `tensor_info`, but since parentheses are special characters in regex, we would have to replace the parentheses with `\(` and `\)`. Do you think it would be worth it?

Reply:
In that case you can call [`re.escape`](https://docs.python.org/3/library/re.html#re.escape) on the `tensor_info` string output to do those substitutions for you.

Reply:
I see. Added the `tensor_info` string in 36dfde0e178c788504a33ab7913cc0711549738c.